pkg/quarkus/v1alpha/api_test.go

```diff
@@ -0,0 +1,155 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/pflag"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+)
+
+var _ = Describe("v1", func() {
+	testAPISubcommand := &createAPISubcommand{}
+	testResource := resource.Resource{
+		GVK: resource.GVK{
+			Group:   "test-group",
+			Version: "v1",
+			Kind:    "Test-Kind",
+		},
+		Plural: "test-plural",
+	}
+
+	Describe("UpdateResource", func() {
+		testAPIOptions := &createAPIOptions{
+			CRDVersion: "testVersion",
+			Namespaced: true,
+		}
+		updateTestResource := resource.Resource{}
+		testAPIOptions.UpdateResource(&updateTestResource)
+
+		It("verify that resource API's CRDVersion was set", func() {
+			Expect(updateTestResource.API.CRDVersion, testAPIOptions.CRDVersion)
+		})
+
+		It("verify that resource API's Namespaced was set", func() {
+			Expect(updateTestResource.API.Namespaced, testAPIOptions.Namespaced)
+		})
+
+		It("verify that resource path is an empty string", func() {
+			Expect(updateTestResource.Path, "")
+		})
+
+		It("verify that resource controller is false", func() {
+			Expect(updateTestResource.Controller).To(BeFalse())
+		})
+	})
+
+	Describe("BindFlags", func() {
+		flagTest := pflag.NewFlagSet("testFlag", -1)
+		testAPISubcommand.BindFlags(flagTest)
+
+		It("should set SortFlags to false", func() {
+			Expect(flagTest.SortFlags).To(BeFalse())
+		})
+
+		It("should set CRDVersion to v1", func() {
+			Expect(testAPISubcommand.options.CRDVersion, "v1")
+		})
+
+		It("should set Namespaced to true", func() {
+			Expect(testAPISubcommand.options.Namespaced).To(BeTrue())
+		})
+	})
+
+	Describe("InjectConfig", func() {
+		testConfig, _ := config.New(config.Version{Number: 3})
+		err := testAPISubcommand.InjectConfig(testConfig)
+
+		It("should set config", func() {
+			Expect(testAPISubcommand.config, testConfig)
+		})
+
+		It("should return nil", func() {
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Describe("Run", func() {
+		It("should return nil", func() {
+			Expect(testAPISubcommand.Run(machinery.Filesystem{})).To(BeNil())
+		})
+	})
+
+	Describe("Validate", func() {
+		It("should return nil", func() {
+			Expect(testAPISubcommand.Validate()).To(BeNil())
+		})
+	})
+
+	Describe("PostScaffold", func() {
+		It("should return nil", func() {
+			Expect(testAPISubcommand.PostScaffold()).To(BeNil())
+		})
+	})
+
+	Describe("InjectResource", func() {
+		failAPISubcommand := &createAPISubcommand{}
+		failResource := resource.Resource{
+			GVK: resource.GVK{
+				Group:   "Fail-Test-Group",
+				Version: "test-version",
+				Kind:    "test-kind",
+			},
+			Plural: "test-plural",
+		}
+
+		groupErr := failAPISubcommand.InjectResource(&failResource)
```

Comment:
These calls should be included with the `It` blocks. Basically consider the `It` as a specific test and it should include the call and the expectations.